torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -643,7 +607,7 @@ def _sfdp_replacement_21(query, key, value, attn_mask):
         query,
         key,
         value,
-        attn_mask=attn_mask,
+        attn_mask=attn_mask.to(dtype=query.dtype),
```

Comment:
As reduced q/k/v and fp32 mask are supported in SDPA, why need type conversion here?

Reply:
Followed other patterns defined in this file. Add type conversion here to make it more robust, because cuda `EFFICIENT_ATTENTION` does not support mixed types of mask and q/k/v.